validator/rpc/apimiddleware/endpoint_factory.go

```diff
@@ -41,6 +41,8 @@ func (*ValidatorEndpointFactory) Create(path string) (*apimiddleware.Endpoint, e
 	case "/eth/v1/validator/{pubkey}/feerecipient":
 		endpoint.GetResponse = &getFeeRecipientByPubkeyResponseJson{}
 		endpoint.PostRequest = &setFeeRecipientByPubkeyRequestJson{}
+	case "/eth/v1/validator/{pubkey}/gaslimit":
```

Comment:
you forgot to add it in the paths function. 
please try to curl request or postman script test this.

you can test the easiest by building the application from the cmd package
1. cd `cmd/validator`
2. `go build -o ./tmp/validator`
3. `./tmp/validator --web ---validators-external-signer-url=http://localhost:9000/` /*this triggers running the validator client in web3signer mode which does not require a prysm wallet*/
4. grab the bearer token and then you can test using postman or curl

Reply:
Thanks for sharing the recipe, this makes things easier. Now following your recipe, I tested the server with:

curl  -H "Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.e30.C-BuVfWhKpr9elB05GTJdEEx_8AzkImxzXL03IqcHR8" 127.0.0.1:7500/eth/v1/validator/0xb4844195ce8ca78d9d4f7ffdf4001cfdb079e059542bcc4f45ddfff2bcec7defb1482db4f9426f92f59972da395dd2b5/gas_limit

And got:
{"data":{"pubkey":"0xb4844195ce8ca78d9d4f7ffdf4001cfdb079e059542bcc4f45ddfff2bcec7defb1482db4f9426f92f59972da395dd2b5","gas_limit":"30000000"}}